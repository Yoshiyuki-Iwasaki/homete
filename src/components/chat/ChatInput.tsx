import firebase from '../../firebase/clientApp';
import { useState } from 'react';
import { useAuthState } from 'react-firebase-hooks/auth';
import styled from 'styled-components';
import { pc, sp, tab } from '../media';

interface Props {
  id:number;
}

const ChatInput = ({ id }: Props) => {
  const db = firebase.firestore();
  const [text, setText] = useState<string>('');
  const [user, userLoading, userError] = useAuthState(firebase.auth());
  const convertJST = new Date();
  convertJST.setHours(convertJST.getHours());
  const updatedTime = convertJST.toLocaleString('ja-JP').slice(0, -3);

  const handleSubmit = async (
    e: React.FormEvent<HTMLFormElement> | React.MouseEvent<HTMLInputElement>,
  ): Promise<string> => {
    e.preventDefault();
    if (!text) return;
    await db.collection('chat').add({
      id: new Date().getTime(),
      message: text,
      userId: user.uid,
      groupeId: id,
      createdAt: updatedTime,
    });
    setText('');
  };

  if (userLoading) {
    return <h6>Loading...</h6>;
  }

  if (userError) {
    return null;
  }

  const Form = styled.form`
    text-align: center;
  `;
  const Input = styled.input`
    width: calc(100% / 3);
    height: 200px;
    border: 1px solid gray;

    ${sp`
      width: 70%;
      height: 90px;
    `}
  `;
  const Button = styled.button`
    margin-left: 20px;
    padding: 5px 10px;
    border: 1px solid gray;
    color: #fff;
    transition: all 0.6s;
    cursor: pointer;
  `;

  return (
    <Form onSubmit={(e) => handleSubmit(e)}>
      <Input
        data-testid="input"
        type="text"
        value={text}
        onChange={(e) => setText(e.target.value)}
        placeholder="みんなに褒めてもらいましょう(^^)"
      />
      <Button data-testid="createButton" onClick={(e) => handleSubmit(e)}>
        投稿する
      </Button>
    </Form>
  );
};

export default ChatInput;
