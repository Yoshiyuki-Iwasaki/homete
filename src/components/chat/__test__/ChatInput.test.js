import { render, fireEvent, cleanup } from '@testing-library/react';
import React from 'react';
import ChatInput from '../ChatInput';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      asPath: '/',
    };
  },
}));

describe('ChatInput 初期表示の確認', () => {
  test('ChatInput', () => {
    const { asFragment } = render(<ChatInput />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders the correct initial DOM', () => {
    const doc = render(<ChatInput />);
    const inputElement = doc.getByTestId('input');
    // The input should be blank.
    expect(inputElement.getAttribute('value')).toBe('');
  });

  test('it creates a new todo', () => {
    const doc = render(<ChatInput />);

    const inputElement = doc.getByTestId('input');
    const createButton = doc.getByTestId('createButton');

    fireEvent.change(inputElement, { target: { value: 'ChatItem.' } });
    fireEvent.click(createButton);

    const todos = doc.queryAllByTestId('todo');
    const todoInput = doc.queryByTestId('todoInput');

    // The name should be in the document as "Feed my dog."
    // expect(todoInput.value).toBe('ChatItem.');

    // expect(inputElement.value).toBe('');

    // // The todo should be in the document.
    // expect(todoInput).toBeInTheDocument();

    // // There should be 1 todo in the document.
    // expect(todos.length).toBe(1);
  });
});
