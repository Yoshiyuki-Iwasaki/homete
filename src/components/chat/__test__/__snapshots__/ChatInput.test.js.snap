// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput 初期表示の確認 ChatInput 1`] = `
<DocumentFragment>
  <form
    class="ChatInput__Form-sc-1obudsn-0 hCcKvs"
  >
    <input
      class="ChatInput__Input-sc-1obudsn-1 huRCKZ"
      data-testid="input"
      placeholder="みんなに褒めてもらいましょう(^^)"
      type="text"
      value=""
    />
    <button
      class="ChatInput__Button-sc-1obudsn-2 cGUHJr"
      data-testid="createButton"
    >
      投稿する
    </button>
  </form>
</DocumentFragment>
`;
